src/chapters/databases-part2/game-logic.rst

```diff
@@ -0,0 +1,161 @@
+Game Page Logic
+===============
+
+Our home page is ready, the database functions operate, and we have access to
+ready-made functions in ``game_logic.py``. After we add one final bit of code,
+we can finally play our Minesweeper game!
+
+The ``play()`` Function
+-----------------------
+
+The ``play()`` function controls the game page. It renders the ``mines.html``
+template, collects and validates data from the form, calls a function to
+analyze each guess, and redirects players to the home page as needed.
+
+Open the ``main.py`` file in Visual Studio Code, then code along with each of
+the following videos.
+
+Collect and Validate Guesses
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. raw:: html
+
+   <section class="vid_box">
+      <iframe class="vid" src="https://www.youtube-nocookie.com/embed/aDLCNfNkS5U" frameborder="1" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+   </section>
+
+Collect Checkbox Data and Call ``check_guess()``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. raw:: html
+
+   <section class="vid_box">
+      <iframe class="vid" src="https://www.youtube-nocookie.com/embed/P_P5l7GQ_9Y" frameborder="1" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+   </section>
+
+``play()`` Summary
+------------------
+
+#. By typing the URL of the game page into the address bar, players can access
+   the board before hiding any mines. To prevent this, we redirect them back to
+   the home page if they arrive at the page through a ``GET`` request.
+#. We need to collect two pieces of information from the game board: which cell
+   the player selected and whether or not they clicked the *Flag Mine*
+   checkbox.
+
+   a. Each button on the board has the same name, and the value for each one
+      matches its cell coordinates (like ``A10``).
+   b. However, the value for the *New Game* button is the empty string.
+   c. The checkbox input has a different name from the buttons.
```

Comment:
I enjoyed playing around assigning different strings

Reply:
Tinkering is always fun!